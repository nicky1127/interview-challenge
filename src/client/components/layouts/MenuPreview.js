import React from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import MenuItem from "../abstractions/MenuItem";
import { filterItemsById } from "../../helpers";

const MenuPreview = props => {
  const { menuItems } = props;
  let content;

  if (Array.isArray(menuItems) && menuItems.length > 0) {
    content = menuItems.map((item, index) => (
      <MenuItem key={`menuItem-${index}`} item={item} idx={index} />
    ));
  }

  return (
    <div data-testid="menuPreview">
      <h2>Menu Preview</h2>
      <ul className="menu-preview">{content}</ul>
    </div>
  );
};

MenuPreview.defaultProps = { menuItems: [] };

MenuPreview.propTypes = {
  menuItems: PropTypes.array
};

const mapStateToProps = state => {
  const { items, selectedItemIDs } = state;
  const menuItems = filterItemsById(items,selectedItemIDs) ;
  return {
    menuItems
  };
};

const ConnectedMenuPreview = connect(mapStateToProps)(MenuPreview);

export default ConnectedMenuPreview;
