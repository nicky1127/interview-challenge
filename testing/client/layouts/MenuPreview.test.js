import { render, cleanup} from "@testing-library/react";
import MenuPreview from "client/components/layouts/MenuPreview";

beforeEach(() => {
  jest.clearAllMocks();
  cleanup();
});


describe("MenuPreview", () => {
  it("Snapshot matchs", () => {
    const tree = renderer
      .create(
          <MenuPreview />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
    it("MenuPreview should render one [data-testid=menuPreview]", () => {
      const { getByTestId } = render(
          <MenuPreview />
      );
      const container = getByTestId("menuPreview");
      expect(container).toBeInTheDocument();
    });
});
