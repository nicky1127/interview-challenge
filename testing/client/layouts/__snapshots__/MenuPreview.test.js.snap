// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuPreview Snapshot matchs 1`] = `
<div
  data-testid="menuPreview"
>
  <h2>
    Menu preview
  </h2>
  <ul
    className="menu-preview"
  >
    <li
      className="item"
    >
      <h2>
        Dummy item
      </h2>
      <p>
        <span
          className="dietary"
        >
          ve
        </span>
        <span
          className="dietary"
        >
          v
        </span>
        <span
          className="dietary"
        >
          n!
        </span>
      </p>
      <button
        className="remove-item"
      >
        x
      </button>
    </li>
    <li
      className="item"
    >
      <h2>
        Dummy item
      </h2>
      <p>
        <span
          className="dietary"
        >
          ve
        </span>
        <span
          className="dietary"
        >
          v
        </span>
        <span
          className="dietary"
        >
          n!
        </span>
      </p>
      <button
        className="remove-item"
      >
        x
      </button>
    </li>
    <li
      className="item"
    >
      <h2>
        Dummy item
      </h2>
      <p>
        <span
          className="dietary"
        >
          ve
        </span>
        <span
          className="dietary"
        >
          v
        </span>
        <span
          className="dietary"
        >
          n!
        </span>
      </p>
      <button
        className="remove-item"
      >
        x
      </button>
    </li>
    <li
      className="item"
    >
      <h2>
        Dummy item
      </h2>
      <p>
        <span
          className="dietary"
        >
          ve
        </span>
        <span
          className="dietary"
        >
          v
        </span>
        <span
          className="dietary"
        >
          n!
        </span>
      </p>
      <button
        className="remove-item"
      >
        x
      </button>
    </li>
  </ul>
</div>
`;
